crates/librqbit/src/blocklist.rs

```diff
@@ -0,0 +1,339 @@
+            Box::pin(BufReader::new(GzipDecoder::new(reader)))
+        } else {
+            trace!("Plain text file detected.");
+            Box::pin(reader)
+        };
+
+        let reader = BufReader::new(reader);
+        let mut lines = reader.lines();
+        let mut ipv4_ranges: Vec<std::ops::Range<u32>> = Vec::new();
+        let mut ipv6_ranges: Vec<std::ops::Range<u128>> = Vec::new();
+        while let Some(line) = lines.next_line().await? {
+            // Skip comments and empty lines
+            if line.starts_with('#') || line.trim().is_empty() {
+                continue;
+            }
+
+            // Parse IP ranges in format: "RuleName:StartIp-EndIp"
+            if let Some((start_ip, end_ip)) = parse_ip_range(&line) {
+                match (start_ip, end_ip) {
+                    (IpAddr::V4(start), IpAddr::V4(end)) => {
+                        let start_num = u32::from_be_bytes(start.octets());
+                        let end_num = u32::from_be_bytes(end.octets());
+                        let range = if end_num == u32::MAX {
+                            start_num..end_num // Special case: Use inclusive range when max
+                        } else {
+                            start_num..(end_num + 1) // Normal case
+                        };
+                        ipv4_ranges.push(range);
+                    }
+                    (IpAddr::V6(start), IpAddr::V6(end)) => {
+                        let start_num = u128::from_be_bytes(start.octets());
+                        let end_num = u128::from_be_bytes(end.octets());
+                        let range = if end_num == u128::MAX {
+                            start_num..end_num // Special case: Use inclusive range when max
+                        } else {
+                            start_num..(end_num + 1) // Normal case
+                        };
+                        ipv6_ranges.push(range);
+                    }
+                    _ => {
+                        continue;
+                    }
+                }
+            }
+        }
+
+        info!(
+            ipv6_entry_count = ipv6_ranges.len(),
+            ipv4_entry_count = ipv4_ranges.len(),
+            "Finished loading blocklist"
+        );
+
+        let blocklist = Self::new(&ipv4_ranges, &ipv6_ranges);
+        Ok(blocklist)
+    }
+
+    pub fn is_blocked(&self, ip: &IpAddr) -> bool {
+        match ip {
+            IpAddr::V4(ipv4) => {
+                let num = u32::from_be_bytes(ipv4.octets());
+                self.ipv4_ranges.query_point(num).next().is_some()
+            }
+            IpAddr::V6(ipv6) => {
+                let num = u128::from_be_bytes(ipv6.octets());
+                self.ipv6_ranges.query_point(num).next().is_some()
+            }
+        }
+    }
+}
+
+fn parse_ip_range(line: &str) -> Option<(IpAddr, IpAddr)> {
+    // Skip comments and empty lines
+    if line.starts_with('#') || line.trim().is_empty() {
```

Comment:
how about

```
let line = line.trim();  // or maybe trim_end()
if line.starts_with('#') || line.is_empty()
```

so that it's more bulletproof against trailing whitespace